### PR TITLE
duplicate and rename hdu rgl prio

### PR DESCRIPTION
**Pull request by buthanoid**

### CODE REVIEW

File: src/main/java/fr/jmmc/oimaging/model/IRModel.java [hunk 1]
```diff
@@ -402,6 +402,15 @@ private void selectHDUs() {
             oifitsFile.getFitsImageHDUs().add(selectedInputImageHDU);
         }
         if (selectedRglPrioImage != null && selectedRglPrioImage != NULL_IMAGE_HDU) {
+
+            // copying and renaming rgl_prio's hdu if it is the same of init_img
+            if (selectedInputImageHDU == selectedRglPrioImage) {
+                selectedRglPrioImage = new FitsImageHDU(selectedInputImageHDU);
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oimaging/model/IRModel.java:408]: use local variable to not alter the user selection = selectedRglPrioImage !

File: src/main/java/fr/jmmc/oimaging/model/IRModel.java [hunk 2]
```diff
@@ -402,6 +402,15 @@ private void selectHDUs() {
...
+                final String newHduName = "PRIO_" + selectedInputImageHDU.getHduName();
+                selectedRglPrioImage.setHduName(newHduName);
+                oifitsFile.getImageOiData().getInputParam().setRglPrio(newHduName);
+            }
+
             oifitsFile.getFitsImageHDUs().add(selectedRglPrioImage);
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oimaging/model/IRModel.java:414]: GUI and input file are now inconsistent !

**Comment by bourgesl** [comment]
That's too painful. I tested and BSMEM works with duplicated HDU (same HDU_NAME), let's wait.

**Comment by bourgesl** [comment]
Finally I prefer the 2nd solution: when calling the IR software, the input OIFits structure in memory is written to a temporary file (before sending it through http) so it seems the good place to fix the generated OIFITS file to be really valid (FITS + OIFITS + Fits image validation) and the writeOIFitsFile() can pre-process / adjust the generated nom.tam Fits HDU ...

See `IRModel.prepareTempFile()` to fix the oifitsFile given to `OIFitsWriter.writeOIFits(tmpFile.getAbsolutePath(), oifitsFile);`

**Comment by buthanoid** [comment]
It was not difficult to do it in `selectHDUs` since it was only testing if the `HDU` for `RGL_PRIO` was "the same" from the one for `INIT_IMG`.
Do you think sharing the `HDU` can cause problems in the rest of the software and that it is safer to do it at the very last moment (in `prepareTempFile`) ?
Also if we are to avoid duplicates when we send to `BSMEM` should we rework `OIFITS` file received from `BSMEM` to remove duplicates too ?

**Comment by bourgesl** [comment]
I am just proposing a possible solution to fix the OIFITS file (duplicate HDU + fix HDU_NAME only before writing file) that seems less complicated than fixing the all GUI to handle HDU duplicatton on the fly if INIT_IMG = RGL_PRIOR (same choice) !
To be discussed.

**Comment by gmella** [comment]
Closed and maybe revisited later after a review with software authors organized by @FerreolS

**Pull request closed without merging**